src/librustc_const_eval/eval.rs

```diff
@@ -569,12 +524,21 @@ fn resolve_trait_associated_const<'a, 'tcx: 'a>(
         // when constructing the inference context above.
         match selection {
             traits::VtableImpl(ref impl_data) => {
-                let name = tcx.associated_item(trait_item_id).name;
+                let name = trait_item.name;
                 let ac = tcx.associated_items(impl_data.impl_def_id)
                     .find(|item| item.kind == ty::AssociatedKind::Const && item.name == name);
                 match ac {
-                    Some(ic) => lookup_const_by_id(tcx, ic.def_id, Substs::empty()),
-                    None => default_value,
+                    Some(ic) => {
+                        let substs = tcx.lift(&impl_data.substs).unwrap();
```

Comment:
We can't panic here, but I guess, yeah, it should be using something similar to trans.